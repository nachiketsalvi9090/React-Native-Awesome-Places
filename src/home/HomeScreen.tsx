/**
 * This represents a simple screen-level container component with rendering
 * and styling that is hooked up to the redux store.
 */

import * as React from 'react';
import { connect } from 'react-redux';

import { StyleSheet, ViewStyle, View, Text, TextStyle } from 'react-native';

// Export component without provider for testing purposes
export class Home extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.textContent}>Edit or remove this file.</Text>
      </View>
    );
  }
}

// Connected component is used with Redux store
export const HomeScreen = connect()(Home);

// This helps auto-completion / type safety with `StyleSheet.create`
interface Style {
  container: ViewStyle;
  textContent: TextStyle;
}

// TypeScript hoists variables. We declare the styles here to keep them out of
// the way of the component definition
const styles = StyleSheet.create<Style>({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    paddingBottom: 100,
  },
  textContent: {
    fontSize: 25,
    textAlign: 'center',
  },
});
