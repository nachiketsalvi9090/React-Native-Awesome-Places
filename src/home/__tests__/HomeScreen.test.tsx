import 'react-native';
import * as React from 'react';

// Note: test renderer must be required after react-native.
import { create } from 'react-test-renderer';

import { Home } from '../HomeScreen';

// Using `create` from `react-test-renderer`, it is possible to convert
// the rendered element into a JSON object as shown below. Using that
// object is a simple way to test a rendered component.

it('renders correctly', () => {
  const tree = create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Since `Home` is just a class, it can be tested like any other
// class without the need to render.
it('should not have any props', () => {
  const home = new Home({});
  expect(Object.keys(home.props).length).toEqual(0);
});
