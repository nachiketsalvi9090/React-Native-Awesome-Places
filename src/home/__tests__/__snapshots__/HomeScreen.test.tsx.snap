// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingBottom": 100,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 25,
        "textAlign": "center",
      }
    }
  >
    Edit or remove this file.
  </Text>
</View>
`;
